nncf/common/quantization/quantizer_propagation/graph.py

```diff
@@ -952,6 +952,13 @@ def merge_traverse_fn(curr_node_key: str,
             affecting_pq, prev_node_key = affecting_pq_and_prev_node_key
             curr_node = self.nodes[curr_node_key]
             curr_node_type = curr_node[QuantizerPropagationStateGraph.NODE_TYPE_NODE_ATTR]
+
+            # Skipping traversing through the INTEGER path.
+            if curr_node_key != prev_node_key:
```

Comment:
Please add a test for your changes.

Reply:
I've added the new synthetic model for this case.

Reply:
This should be tested in the common, solver-level tests (`tests/common/quantization/test_quantizer_propagation_solver.py`)

Reply:
Done